trl/trainer/dpo_trainer.py

```diff
@@ -205,7 +205,11 @@ def __init__(
                     "You are using a `peft` version that does not support `disable_adapter`. Please update your `peft` version to the latest version."
                 )
         else:
-            self.ref_model = self.accelerator.prepare_model(self.ref_model, evaluation_mode=True)
+            if self.is_deepspeed_enabled:
+                (self.ref_model,) = self.accelerator._prepare_deepspeed(self.ref_model)
```

Comment:
```suggestion
                # Read more about the issue in https://github.com/huggingface/trl/pull/687
                self.ref_model = self.accelerator._prepare_deepspeed(self.ref_model)
```

Reply:
@younesbelkada 
good idea!
i agree this!

Reply:
Great! Feel free to accept the suggestion above